### DISCUSSION TITLE
How can I delete a message?

### DISCUSSION THREAD

**Discussion by yutanov** [python-telegram-bot/python-telegram-bot#3074] [Q&A] [answered]
Please tell me the solution.
When I'm trying to delete message:

```
from telegram import Update, messageid

context.bot.delete_message(chat_id=Update.effective_chat.id, message_id=messageid.MessageId)
```

I get an error:

>   File "/usr/lib/python3.10/json/__init__.py", line 231, in dumps
>     return _default_encoder.encode(obj)
>   File "/usr/lib/python3.10/json/encoder.py", line 199, in encode
>     chunks = self.iterencode(o, _one_shot=True)
>   File "/usr/lib/python3.10/json/encoder.py", line 257, in iterencode
>     return _iterencode(o, 0)
>   File "/usr/lib/python3.10/json/encoder.py", line 179, in default
>     raise TypeError(f'Object of type {o.__class__.__name__} '
> TypeError: Object of type type is not JSON serializable
> 

What is wrong?

**Comment by harshil21** [comment, marked as answer]
Hello,

You can only use `context` inside a callback function, like [here](https://github.com/python-telegram-bot/python-telegram-bot/blob/v13.x/examples/pollbot.py#L83). Also, there's no import called as `messageid` in our library. The correct call would be:

``` py
def callback(update: Update, context: CallbackContext):
    context.bot.delete_message(chat_id=update.effective_chat.id, message_id=update.effective_message.id)
```

Also, if you use PTB's shortcuts, the above boils down to:
``` py
def callback(update: Update, context: CallbackContext):
    update.effective_message.delete()
```

**Reply by yutanov** [reply]
Thank you!
I was able to delete the message with:

`update.effective_message.delete()`

But when i tried to remove it with:

`context.bot.delete_message(chat_id=update.effective_chat.id, message_id=update.effective_message.id)`

I got an error:

> 
AttributeError: 'Message' object has no attribute 'id'

**Reply by Bibo-Joshi** [reply]
it's `message_id`

**Reply by yutanov** [reply]
Thank you so much! It works.
